pkg/storaged/stratis/pool.jsx

```diff
@@ -64,6 +65,283 @@
 const fsys_min_size = 512 * 1024 * 1024;
 const fsys_round_size = 1024 * 1024;
 
+/* Abstractions over the r6 and r8 API revisions, and V1 and V2 pool
+ * formats.
+ */
+
+/* Key descriptions are returned as an array with elements of type
+ *
+ * { slot: number | null, keydesc: string }
+ *
+ * A V1 pool will have at most one entry in the array, and it will
+ * have `slot == null`.  V2 pools have zero or more entries with
+ * numbered slots.
+ */
+
+function get_key_descriptions(pool) {
+    const result = [];
+
+    if (!pool.Encrypted)
+        return result;
+
+    if (client.stratis_interface_revision < 8) {
+        const key_desc = pool.KeyDescription;
+        /* The KeyDescription property has type "(b(bs))".
+
+           The first "b" is false when the encryption of the pool is
+           inconsistent (not all block devices have the same LUKS
+           header). We pretend that the pool doesn't have any
+           passphrase in that case.
+
+           The second "b" is true when the pool has a passphrase, and
+           the "s" is the description of the key for it.
+
+           Thus, we detect an object of shape
+
+               [ true, [ true, "..." ] ]
+
+           and fish the string out of it.
+         */
+        if (key_desc[0] && key_desc[1][0])
+            result.push({ slot: null, keydesc: key_desc[1][1] });
+    } else {
+        let key_descs = pool.KeyDescriptions;
+        /* The KeyDescriptions property itself has type "v". For a V1
+           pool this variant carries a value of type "b(bs)" just as
+           with the r6 API (see above). For a V2 pool, it carries a
+           "a(us)" value, an array of "slot number plus key
+           description".
+
+           Since the property itself has type "v", we need unwrap this
+           variant explicitly.
+
+           HACK - https://github.com/stratis-storage/stratisd/issues/3895
+           However, change notifications drop the variant
+           wrapping for some reason, so we only do the unwrapping when
+           "key_descs" is indeed a variant.
+         */
+        if ("t" in key_descs)
+            key_descs = key_descs.v;
+        if (pool.MetadataVersion == 1) {
+            if (key_descs[0] && key_descs[1][0])
+                result.push({ slot: null, keydesc: key_descs[1][1] });
+        } else if (pool.MetadataVersion == 2) {
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-21927-4a5fe6bc-20250826-110716-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)